From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Joe Konno <joe.konno@intel.com>
Date: Wed, 31 Jul 2019 10:12:30 -0700
Subject: [PATCH] Settings.c: show thread names by default

This is to support analysis of multi-threaded applications "out of the
box", like web browsers. Applications can name their threads, so this
change should prove useful.

$HOME/.conf/htop/htoprc is written whenever the user changes settings in
htop. If users have a "show_thread_names=0" line in htoprc, this change
will not override it.
---
 Settings.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Settings.c b/Settings.c
index 5ca998a..0bf607f 100644
--- a/Settings.c
+++ b/Settings.c
@@ -448,7 +448,7 @@ Settings* Settings_new(unsigned int initialCpuCount, Hashtable* dynamicColumns)
    this->direction = -1;
    this->treeDirection = 1;
    this->shadowOtherUsers = false;
-   this->showThreadNames = false;
+   this->showThreadNames = true;
    this->hideKernelThreads = true;
    this->hideUserlandThreads = false;
    this->treeView = false;
